contracts/Oracles/ChainlinkPriceOracleWstETH.sol

```diff
@@ -1,29 +1,34 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity 0.8.19;
 
+import { Fixed256x18 } from "@tempusfinance/tempus-utils/contracts/math/Fixed256x18.sol";
 import { IWstETH } from "../Dependencies/IWstETH.sol";
-import { AggregatorV3Interface, BaseChainlinkPriceOracle } from "./BaseChainlinkPriceOracle.sol";
-import { BasePriceOracleWstETH } from "./BasePriceOracleWstETH.sol";
+import { AggregatorV3Interface, ChainlinkPriceOracle } from "./ChainlinkPriceOracle.sol";
+import { IPriceOracleWstETH } from "./Interfaces/IPriceOracleWstETH.sol";
 
-contract ChainlinkPriceOracleWstETH is BaseChainlinkPriceOracle, BasePriceOracleWstETH {
-    // --- Constants ---
+contract ChainlinkPriceOracleWstETH is IPriceOracleWstETH, ChainlinkPriceOracle {
+    using Fixed256x18 for uint256;
 
-    uint256 public constant override DEVIATION = 1e16; // 1%
+    // --- Immutables ---
+
+    IWstETH public immutable override wstETH;
 
     // --- Constructor ---
 
     constructor(
-        AggregatorV3Interface _priceAggregatorAddress,
-        IWstETH _wstETH
+        AggregatorV3Interface priceAggregatorAddress_,
+        IWstETH wstETH_,
+        uint256 deviation_
     )
-        BaseChainlinkPriceOracle(_priceAggregatorAddress)
-        BasePriceOracleWstETH(_wstETH)
-    // solhint-disable-next-line no-empty-blocks
-    { }
-
-    // --- Functions ---
+        ChainlinkPriceOracle(priceAggregatorAddress_, deviation_)
+    {
+        if (address(wstETH_) == address(0)) {
+            revert InvalidWstETHAddress();
+        }
+        wstETH = IWstETH(wstETH_);
+    }
 
     function _formatPrice(uint256 price, uint256 answerDigits) internal view override returns (uint256) {
-        return _convertIntoWstETHPrice(price, answerDigits);
+        return super._formatPrice(price, answerDigits).mulDown(wstETH.stEthPerToken());
```

Comment:
Same as previous ("this part is now duplicated, and earlier, it wasn't (there was abstract class for this) ")